### PR TITLE
added the set up associations

### PR DESCRIPTION
**Pull request by moise10r**
### Create a class Classroom with the following:
 - @Label instance variable, should be initialized in the constructor.
 - Setter and getter for @Label (remember about attr_accessor).
 - Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
 - Create the has-many side (a classroom has many students).
 - Create the belongs-to side (a student belongs to a classroom).
 - Make sure that when adding a student to a classroom it also sets the classroom for the student.
 - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
### Create a class Book with the following:
 - @title and @author instance variables should be initialized in the constructor.
 - Setters and getters for instance variables (remember about attr_accessor).
### Create a class Rental with the following:
 - @Date instance variable should be initialized in the constructor.
 - Setter and getter for @Date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class - -- 
-  - Rental. The following should be implemented:
 - Create the has-many side of Book and Rental (a book has many rentals).
 - Create the belongs-to side of Rental and Book (a rental belongs to a book).
 - Create the has-many side of Person and Rental (a person has many rentals).
 - Create the belongs-to side of Rental and Person (a rental belongs to a person).
 - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (vichuge) [inline book.rb:3]: - [x] Please, add `:rentals` for getter and setter. This variable could be writable too. 🐶

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (vichuge) [inline rental.rb]: - [x] Book and person could be writable too, please change it. 🐱

File: person.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@
 require_relative 'corrector'
 
 class Person
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (vichuge) [inline person.rb:5]: - [x] rentals could be inside accessor too. 🌵

**Review by vichuge — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi dear student, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 
You used proper GitHub Flow ✔️
Your linter checks are all passing ✔️
Your PR messages are descriptive ✔️

## Aspects to improve ♻️

- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (moise10r) [reply book.rb:3]: :rentals were set as editable as well

File: rental.rb (hunk 2 above)
Comment (moise10r) [reply rental.rb]: Issue fixed

File: person.rb (hunk 3 above)
Comment (moise10r) [reply person.rb:5]: Issue fixed

File: book.rb (hunk 1 above)
Comment (moise10r) [reply book.rb:3]: setter

- This is the setter method for rentals

File: rental.rb (hunk 2 above)
Comment (moise10r) [reply rental.rb]: rental
- Add the setter methods for book and person

File: person.rb (hunk 3 above)
Comment (moise10r) [reply person.rb:5]: setter
 - this is the setter method for rentals in person class.

**Review by vichuge — APPROVED** [review]
# Approved project! ✅ 🥳 🎈 🎉🎉🎉🎉🎉🎉🎉🎉 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

InbornCandidHornet-size_restricted

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**